File: JavaScript/12thOct.js
let Arr1 = [0, 1, 2, 0, 1, 2];
let Arr2 = [0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1];

function swap(a, i, j) {
    let temp = a[i];
    a[i] = a[j];
    a[j] = temp;
}

function sortArr(arr, high) {
    let low = 0;
    let mid = 0;
    let base = 1;
    while (mid <= high) {
        if (arr[mid] < base) {
            swap(arr, low, mid);
            ++low; ++mid;
        }
        else if (arr[mid] > base) {
            swap(arr, mid, high)
            --high;
        }
        else {
            ++mid;
        }
    }
}

sortArr(Arr1, Arr1.length - 1)
sortArr(Arr2, Arr2.length - 1)

console.log(Arr1);
console.log(Arr2);

File: Python/12th_oct.py
def sortArr( a, arr_size): 
    lo = 0
    hi = arr_size - 1
    mid = 0
    while mid <= hi: 
        if a[mid] == 0: 
            a[lo], a[mid] = a[mid], a[lo] 
            lo = lo + 1
            mid = mid + 1
        elif a[mid] == 1: 
            mid = mid + 1
        else: 
            a[mid], a[hi] = a[hi], a[mid]  
            hi = hi - 1
    return a 
def Array( a): 
    for k in a: 
        print(k)
arr =list(map(int,input().split(" ")))
arr_size = len(arr) 
arr = sortArr( arr, arr_size) 
Array(arr) 
